cmd/vsphere-csi/main.go

```diff
@@ -58,20 +54,17 @@ func main() {
 	if err != nil {
 		log.Errorf("Failed retrieving cluster flavor. Error: %v", err)
 	}
-	serviceMode := os.Getenv("X_CSI_MODE")
+	serviceMode := os.Getenv(csitypes.EnvVarMode)
 	commonco.SetInitParams(ctx, clusterFlavor, &service.COInitParams, *supervisorFSSName, *supervisorFSSNamespace,
 		*internalFSSName, *internalFSSNamespace, serviceMode)
 
-	gocsi.Run(
```

Comment:
We can introduce an environment variable to determine whether to use gocsi, or use the new code. But the current pipelines have been run with new code and they're passing.